src/core/abr/banned_representations.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import config from "../../config";
+import log from "../../log";
+import { Representation } from "../../manifest";
+
+const { ABR_BAN_TIME_STEPS } = config;
+
+/**
+ * From a given quality, return the ban duration.
+ * @param {number} quality
+ * @returns {number} - ban duration
+ */
+function getBanDurationFromStreamQuality(playbackQuality: number): number {
+  const steps = ABR_BAN_TIME_STEPS.sort((a, b) => a.minimumQuality - b.minimumQuality);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Move this array "sort" operation to a separate statement. ([typescript:S4043](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4043))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=295&issues=AWaxD4f4LpqqSUlcU4f_&open=AWaxD4f4LpqqSUlcU4f_)

[ISSUE_KEY]: AWaxD4f4LpqqSUlcU4f_